src/App.js

```diff
@@ -2,21 +2,29 @@ import React, { useState, useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState(key, initialState) {
-    const savedState = JSON.parse(localStorage.getItem(key)) || initialState;
-    const [state, setState] = useState(savedState);
+function App() { 
+    const key = 'savedTodoList'
+    const savedState = JSON.parse(localStorage.getItem(key)) || [];
```

Comment:
Clearly defining your saved state before passing it to `useState` make this really easy to read. Nice. You can also directly pass `JSON.parse(localStorage.getItem(key)) || []` to `useState` params but this is nice too.